lib/dbutils.js

```diff
@@ -778,16 +800,30 @@ dbu.buildTableSql = function(schema, tableName) {
     return sql;
 };
 
-dbu.buildIndexViewSql = function(schema, tableName) {
-    var grouping = constructGroupingByHashKeys(schema);
-    var order = constructOrder(null, schema);
-    var sql = 'create view if not exists [' + tableName + '_indexView] as ';
-    sql += ' select * from [' + tableName + '_data]';
-    if (isStaticJoinNeeded(null, schema)) {
-        sql += ' natural left outer join [' + tableName + '_static] ';
+dbu.buildSecondaryIndexTableSql = function(schema, tableName) {
+    var result = [];
+    var secondaryIndexNames = Object.keys(schema.secondaryIndexes);
+    var tableSql;
+    if (secondaryIndexNames.length === 0) {
+        return result;
     }
-    sql += grouping + order;
-    return sql;
+
+    tableSql = 'create table if not exists ' + '[' + tableName + '_secondaryIndex] (';
+    schema.allSecondaryIndexKeys.forEach(function(attr) {
+        tableSql += dbu.fieldName(attr) + ' ' + schema.converters[schema.attributes[attr]].type + ', ';
+    });
+    tableSql += 'primary key (' + schema.secondaryIndexPrimaryKeys.map(dbu.fieldName).join(', ') + ') )';
+    result.push(tableSql);
+
+    // Next, create SQLite indexes over secondary index key columns for faster lookup
+    Object.keys(schema.secondaryIndexes).forEach(function(indexName) {
+        var indexSchema = schema.secondaryIndexes[indexName];
+        var indexSql = 'create index if not exists [' + tableName + '_index_' + indexName + ']';
+        indexSql += ' on [' + tableName + '_secondaryIndex] (';
+        indexSql += getAllKeysOfTypes(indexSchema, 'hash', 'range').map(dbu.fieldName).join(', ') + ')';
```

Comment:
@d00rman Fixed. Also added a test https://github.com/wikimedia/restbase-mod-table-spec/pull/7 It's not for this particular issue (we are creating native indexes here, so can't verify it), but for a similar possible issue. Anyway more tests is always better